src/mol-io/reader/_spec/mol2.spec.ts

```diff
@@ -476,6 +726,30 @@ describe('mol2 reader', () => {
         expect(bonds.status_bits.value(0)).toBe('');
     });
 
+    it('atom status_bit', async () => {
+        const parsed = await parseMol2(Mol2AtomWithStatusBit, '').run();
+        if (parsed.isError) {
+            throw new Error(parsed.message);
+        }
+        const mol2File = parsed.result;
+        const data = mol2File.structures[0];
+        const statusBits = data.atoms.status_bit.toArray();
+        expect(statusBits.length).toEqual(1);
+        expect(statusBits[0]).toEqual('BACKBONE|DICT|DIRECT');
+    });
+
+    it('bond status_bit', async () => {
+        const parsed = await parseMol2(Mol2BondWithStatusBit, '').run();
+        if (parsed.isError) {
+            throw new Error(parsed.message);
+        }
+        const mol2File = parsed.result;
+        const data = mol2File.structures[0];
+        const statusBits = data.bonds.status_bits.toArray();
+        expect(statusBits.length).toEqual(1);
+        expect(statusBits[0]).toEqual('BACKBONE|DICT|INTERRES');
```

Comment:
There is a bug where the status bits are assigned to the wrong atom. This is caused by the field not being present in the 1st row, but rather 2nd in case of the atoms, and 18th in case of the bonds.

So the test should say 

```
expect(statusBits.length).toEqual(data.atoms.count);
expect(statusBits[1]).toEqual('BACKBONE|DICT|DIRECT');
for (let i = 0; i < data.atoms.count; i++) {
  if (i !== 1) expect(statusBits[i]).toEqual('');
}
```

(and `statusBits[17]` for the bonds).

To fix this, you need to include

```ts
for (let cI = tokenIndex; cI < tokenIndexToColumn.length; cI++) {
  TokenBuilder.addUnchecked(col, 0, 0); // I hope this works as expected
}
```

to `_readColumnsAndGetMaxCount` at the end of the for loop there. 

The original parsed was written with the assumption that each row has the same number of columns so it was not present.

Reply:
Oh thanks for your help, It seems I missunderstood the data structure of status_bits.

Reply:
Good. Still need to remove the `dataLines` you've added.

Reply:
Got it. Thanks for the heads-up!